pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -131,11 +132,25 @@ public PulsarClientImpl(ClientConfigurationData conf) throws PulsarClientExcepti
     }
 
     public PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopGroup) throws PulsarClientException {
-        this(conf, eventLoopGroup, new ConnectionPool(conf, eventLoopGroup));
+        this(conf, eventLoopGroup, new ConnectionPool(conf, eventLoopGroup), null);
     }
 
     public PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopGroup, ConnectionPool cnxPool)
             throws PulsarClientException {
+        this(conf, eventLoopGroup, cnxPool, null);
+    }
+
+    public PulsarClientImpl(ClientConfigurationData conf, Timer timer) throws PulsarClientException {
```

Comment:
do we need to add so many new public constructors? 
can we add only the needed ones ?

Reply:
> We cannot expose Netty classes on the client API.
> They should construct the PulsarClientImpl, not the ClientBuilder. This is not a common use, just for some application like Pulsar proxy does.
I add so many new public constructors for being compatible with last version, any idea for just add the needed ones?

Reply:
I think we only need to add the last one constructor, with all the arguments. It's not necessary to add timer to all the other constructors.

Reply:
ok